lib/_stream_writable.js

```diff
@@ -273,13 +273,14 @@ Writable.prototype.write = function(chunk, encoding, cb) {
       cb = nop;
   }
 
-  let err;
   if (state.ending) {
-    err = new ERR_STREAM_WRITE_AFTER_END();
+    throw new ERR_STREAM_WRITE_AFTER_END();
   } else if (state.destroyed) {
-    err = new ERR_STREAM_DESTROYED('write');
+    // TODO(ronag): Compat. Throwing here can break stuff.
+    // throw new ERR_STREAM_DESTROYED('write');
```

Comment:
Throwing here can break `Readable.pipe` since it doesn't automatically `unpipe` on error nor does it check `writable` or `destroyed` before calling `write()`.